### PR TITLE
Refactor MTE-1000 [v115] Rebalance smoke tests

### PR DESCRIPTION
**Pull request by clarmso**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-1000)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

### Description
Smoke test 3 is taking longer than others to execute. Let's redistribute the tests so that we can have faster response time in the pipeline.

### Pull requests checks where applicable
- [ ] Fill in the three TODOs above (tickets number and description of your work)
- [X] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -26,17 +27,6 @@ workflows:
...
-        title: Bitrise Test Plan Sharder
-        inputs:
-        - xcode_project: Client.xcodeproj
-        - target: Client
-        - shards: '3'
-        - scheme: Fennec
-        - test_plan: SmokeXCUITests.xctestplan
-        - debug_mode: 'true'
-        - test_path: ''
-        - file_type: ".swift"
```
Comment (clarmso) [inline bitrise.yml:39]: I have tried using this step to divide up the test plan, but I didn't get it working for me. In addition, this step has been error and is not used, so let's remove this step entirely.

**Comment by clarmso** [comment]
5 smoke tests test plans. 18m11s. 208 credits. 🤔 
https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/b7237079-5935-4529-8872-d9323d73621e

We may spend more credits by having 5 test plans. Here's the result with 4 test plans.
https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/cda99857-490f-4e16-99fe-1de8389d92a9

**Comment by isabelrios** [comment]
> 5 smoke tests test plans. (Duration: 18m 11s. Credits: 208) 🤔 https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/b7237079-5935-4529-8872-d9323d73621e
> 
> We may spend more credits by having 5 test plans. Here's the result with 4 test plans. (Duration: 18m57s. Credits: 196) https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/cda99857-490f-4e16-99fe-1de8389d92a9

Thanks for doing this exploration. You are right, looking at the numbers look like we don't actually save significant time and we spend more credits when using 5 test plans. 
If we can keep the xcode test parallelization, let's keep our 4 test plan. Though, we need to balance the Smoketest3 as you said.

**Comment by clarmso** [comment]
> Thanks for doing this exploration. You are right, looking at the numbers look like we don't actually save significant time and we spend more credits when using 5 test plans. If we can keep the xcode test parallelization, let's keep our 4 test plan. Though, we need to balance the Smoketest3 as you said.

Let me create a new PR since the work for MTE-999 has been merged. The new PR should have a test or two moving over from `Smoketest3` to `Smoketest4`.

**Comment by isabelrios** [comment]
> > Thanks for doing this exploration. You are right, looking at the numbers look like we don't actually save significant time and we spend more credits when using 5 test plans. If we can keep the xcode test parallelization, let's keep our 4 test plan. Though, we need to balance the Smoketest3 as you said.
> 
> Let me create a new PR since the work for MTE-999 has been merged. The new PR should have a test or two moving over from `Smoketest3` to `Smoketest4`.

Actually the latest PR shows that the difference is not that big among test suites
Screenshot 2023-05-19 at 14 06 34
https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/fe8c1a64-e65a-44dd-acd5-55d168cb8a43

Let's monitor the builds and see how we can make this right

**Pull request closed without merging**